### PR TITLE
Add get_device_from_id and get_device_from_array

### PR DESCRIPTION
**Pull request by mitmul**
Follow up of this PR: https://github.com/pfnet/chainer/pull/2243

### CODE REVIEW

**Comment by okuta** [comment]
Please add tests for new functions.

File: chainer/cuda.py [hunk 1]
```diff
@@ -150,9 +150,47 @@ def __ne__(self, other):
...
 # Global states
 # ------------------------------------------------------------------------------
+def get_device_from_id(device_id):
+    """Gets the device from an ID integer.
+
+    Args:
+        device_id (int or None): The ID of the device which this function
+            returns.
+    """
+    if type(device_id) in _integer_types:
```
Comment (mitmul) [inline chainer/cuda.py]: @unnonouno Thank you for your suggestion. I fixed it.

**Comment by okuta** [comment]
@mitmul Cloud you add tests for these new functions.

**Comment by mitmul** [comment]
@okuta Thanks for your comment. I added tests for those two functions.

**Comment by mitmul** [comment]
Finally... : https://ci-chainer.pfidev.jp/job/chainer-sakura/1249/

File: tests/chainer_tests/test_cuda.py [hunk 2]
```diff
@@ -43,6 +44,30 @@ def test_get_dummy_device_for_empty_array(self):
         self.assertIs(cuda.get_device(x), cuda.DummyDevice)
 
     @attr.gpu
+    def test_get_device_warning(self):
+        # NOTE: Python2.7 has a bug in catch_warnings, so this test is skipped
+        # for Python2.7
+        if six.PY3:
```
Comment (okuta) [inline tests/chainer_tests/test_cuda.py]: Please use `@nittest.skipUnless`

File: chainer/cuda.py [hunk 3]
```diff
@@ -153,9 +153,47 @@ def __ne__(self, other):
...
+            return array.device
+    return DummyDevice
+
+
 def get_device(*args):
     """Gets the device from a device object, an ID integer or an array object.
 
+    .. note::
+
+        This API is deprecated. Please use :method:`cuda.get_device_from_id`
```
Comment (unnonouno) [inline chainer/cuda.py]: `cupy.cuda.get_device_from_id`?

File: chainer/cuda.py [hunk 4]
```diff
@@ -153,9 +153,47 @@ def __ne__(self, other):
...
+        or :method:`cuda.get_device_from_array` instead.
```
Comment (unnonouno) [inline chainer/cuda.py]: ditto

File: chainer/cuda.py [hunk 5]
```diff
@@ -153,9 +153,47 @@ def __ne__(self, other):
...
+    The device on which the given CuPy array reside is returned.
+
+    Args:
+        array (:class:`cupy.ndarray` or list of :class:`cupy.ndarray`):
+            A CuPy array which this function returns the device corresponding
+            to. If a list of :class:`cupy.ndarray`s are given, it returns
+            the first device object of an array in the list.
+    """
+    for array in arrays:
+        if isinstance(array, ndarray):
```
Comment (unnonouno) [inline chainer/cuda.py]:
```
if isinstance(array, ndarray) and array.device is not None:
    return array.device
```

**Comment by mitmul** [comment]
Fixed

**Comment by okuta** [comment]
OK

**Pull request merged**